src/Eccube/Controller/InstallPluginController.php

```diff
@@ -51,57 +107,65 @@ public function pluginEnable(SystemService $systemService, PluginService $plugin
             try {
                 ob_start();
 
-                $pluginService->installWithCode($Plugin->getCode());
+                if ($Plugin->isEnabled()) {
+                    $pluginService->disable($Plugin);
+                } else {
+                    $pluginService->enable($Plugin);
+                }
 
-                $pluginService->enable($Plugin);
             } finally {
                 $log = ob_get_clean();
                 while (ob_get_level() > 0) {
                     ob_end_flush();
                 }
             }
-        }
 
-        return $this->json(['success' => true, 'log' => $log]);
+            $this->cacheUtil->clearCache();
+            return $this->json(['success' => true, 'log' => $log]);
+        } else {
+            return $this->json(['success' => false, 'log' => $log]);
+        }
     }
 
     /**
-     * プラグインを有効にします。
+     * トランザクションファイルを削除し, 管理画面に遷移します.
      *
-     * @Route("/install/cache/clear", name="install_cache_clear")
+     * @Route("/install/plugin/redirect", name="install_plugin_redirect")
      *
-     * @return JsonResponse
+     * @return RedirectResponse
      */
-    public function cacheClear(CacheUtil $cacheUtil)
+    public function redirectAdmin()
     {
-        // トランザクションチェックファイルの有効期限を確認する
-        if (!$this->isValidTransaction()) {
-            throw new NotFoundHttpException();
-        }
-
-        $cacheUtil->clearCache();
+        $this->cacheUtil->clearCache();
 
         // トランザクションファイルを削除する
         $projectDir = $this->getParameter('kernel.project_dir');
-        unlink($projectDir.parent::TRANSACTION_CHECK_FILE);
+        $transaction = $projectDir.parent::TRANSACTION_CHECK_FILE;
+        if (file_exists($transaction)) {
+            unlink($transaction);
+        }
 
-        return $this->json(['success' => true]);
+        return $this->redirectToRoute('admin_homepage');
```

Comment:
無駄なリダイレクトが発生するのでログイン画面の方が良いと思います。
```suggestion
        return $this->redirectToRoute('admin_login');
```